### PR TITLE
Cleanup QgsDistanceArea API and sipify, add cache for ellipsoid lookup

### PR DESCRIPTION
**Pull request by nyalldawson**
Some 3.0 cleanups for the QgsDistanceArea api...

### CODE REVIEW

**Review by wonder-sk — APPROVED** [review]

**Comment by nyalldawson** [comment]
@wonder-sk I've added a cache for the ellipsoid lookup - can I request a re-review please? :)

**Review by wonder-sk — COMMENTED** [review]
Caching of ellipsoids - oh yeah! Looks good to me, with one small note below...

I am wondering if this would be a good time to also throw away ```setEllipsoidalMode()``` method :-) Seems that calling ```setEllipsoid()``` is already good enough to distinguish whether we want planar or geodetic distances/areas

File: src/core/qgsdistancearea.cpp [hunk 1]
```diff
@@ -118,22 +85,35 @@ void QgsDistanceArea::setSourceAuthId( const QString &authId )
...
     return true;
   }
 
+  // check cache
+  sEllipsoidCacheLock.lockForRead();
+  QHash< QString, EllipsoidParameters >::const_iterator cacheIt = sEllipsoidCache.constFind( ellipsoid );
+  if ( cacheIt != sEllipsoidCache.constEnd() )
+  {
+    // found a match in the cache
+    sEllipsoidCacheLock.unlock();
```
Comment (nyalldawson) [inline src/core/qgsdistancearea.cpp]: Thanks - fixed!

**Comment by nyalldawson** [comment]
> I am wondering if this would be a good time to also throw away setEllipsoidalMode() method :-) Seems that calling setEllipsoid() is already good enough to distinguish whether we want planar or geodetic distances/areas

Done, with a lot of other API cleanups too.

**Pull request merged**